### DISCUSSION TITLE
Intermittent issue with first request?

### DISCUSSION THREAD

**Discussion by edlea** [timdorr/tesla-api#279] [Q&A] [answered]
I am seeing fairly consistent, but intermittent, issues with the first request to `https://auth.tesla.com/oauth2/v3/authorize?...`

A reasonable number of responses (I'd got as far as saying 50%) do not have the HTML form; it's as if someone at Tesla.com has only deployed to one instance of two behind a load balancer.

I am seeing this with my own Dart implementation and [teslajs](https://github.com/mseminatore/TeslaJS). Are other people seeing the same problem or is a shared implementation issue?

**Comment by jasminsehic** [comment, marked as answer]
There is a WAF in place, try backing off a little while and it should work ok.

**Reply by insaneoctane** [reply]
Define "back off a little"?  Same problem here.  Very difficult to troubleshoot my script when I can't reliably get past step 1!  I ran successfully last night at midnight, but can't get a single pass through 12 hours later....

Is it user-agent?  Is it (optional) logon_hint?  Driving me nuts!

**Reply by briangru** [reply]
One note on the user agent - I found that I had to specify a user agent with a period in the string, or making requests would hang.  Sounds like a different problem, but I thought I'd try providing a little certainty on insane octane's question.
